server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java

```diff
@@ -228,8 +228,9 @@ private void allocateUnassigned() {
                 }
 
                 final var allocationStatus = UnassignedInfo.AllocationStatus.fromDecision(
-                    isThrottled ? Decision.Type.THROTTLE : Decision.Type.NO
+                    isThrottled || desiredNodeIds.isEmpty() ? Decision.Type.THROTTLE : Decision.Type.NO
```

Comment:
Just realized that this breaks `ClusterAllocationExplainIT#testAllocationFilteringOnIndexCreation` with:
```
expected:<DECIDERS_NO> but was:<DECIDERS_THROTTLED>
Expected :DECIDERS_NO
Actual   :DECIDERS_THROTTLED
```

So we need to distinguish between `desiredNodeIds` not computed / computed but empty. I am working on that.